src/plugins/layout/shadowdom.ts

```diff
@@ -1,10 +1,9 @@
 import { assert, isNumber, traverseNodeTree } from "../../utils";
-import { getNodeIndex, IgnoreTag, NodeIndex } from "./stateprovider";
+import { getNodeIndex, NodeIndex, shouldIgnoreNode } from "./stateprovider";
 
 // Class names for child list mutation classifications
 const FinalClassName = "cl-final";
 const NewNodeClassName = "cl-new";
-const TopNewNodeClassName = "cl-new-top";
 const MovedNodeClassName = "cl-moved";
 const UpdatedNodeClassName = "cl-updated";
```

Comment:
Remove action is processed as moving the node to the 'removedNodes' container. Since every 'Move' is really a remove followed by an insert instantly, we can only deterministically say that node was really removed, when we process all mutations and node ends up in the 'removedNodes' container. Please see the comments I added before the functions for richer details.

---
In reply to: [131237174](https://github.com/Microsoft/clarity-js/pull/25#discussion_r131237174) [](ancestors = 131237174)